chrome/content/otr.js

```diff
@@ -240,6 +240,46 @@ var otr = {
     });
   },
 
+  // forget the private key and instance tag of
+  // the given account and clear them from files.
+  forgetPrivateKey: function(deletedAcc, deletedPrpl) {
+    // clear instance tag
+    let instag = libOTR.otrl_instag_find(this.userstate, deletedAcc, deletedPrpl);
+    if (!instag.isNull()) {
+      libOTR.otrl_instag_forget(instag);
+      if (libOTR.otrl_instag_write(this.userstate, this.instanceTagsPath)) {
+        throw new Error("Failed to write instance tags.");
+      }
+    }
+    // clear private key
+    let sk = libOTR.otrl_privkey_find(this.userstate, deletedAcc, deletedPrpl);
+    if (!sk.isNull()) {
+      libOTR.otrl_privkey_forget(sk);
+    }
+    // hacky
+    let accounts = Services.accounts.getAccounts();
+    if (!accounts.hasMoreElements()) {
+      helpers.removeFile(this.privateKeyPath);
+    } else {
+      for (let acc of helpers.getAccounts()) {
+        let account = acc.normalizedName;
+        let protocol = acc.protocol.normalizedName;
+        let key = libOTR.otrl_privkey_find(this.userstate, account, protocol);
+        if (key.isNull()) continue;
+        let newKey = new libOTR.s_pending_privkey_calc();
+        newKey.accountname = libC.strdup(key.contents.accountname);
+        newKey.protocol = libC.strdup(key.contents.protocol);
+        newKey.privkey = key.contents.privkey;
```

Comment:
I have some troubles with this, would love to hear your ideas here. @arlolra 

I tried with `length = gcry_sexp_sprint(); gcry_sexp_new(newKey.privkey, key.contents.privkey, length, {0,1})` but it didn't work.

Reply:
> I tried with `length = gcry_sexp_sprint();`

I'm assuming you mean `gcry_sexp_length`?

> `gcry_sexp_new(newKey.privkey, key.contents.privkey, length, {0,1})` but it didn't work.

The definition of `gcry_sexp_new` is such that the second argument is expected to be "its external representation" but here `key.contents.privkey` has already been parsed.

I would say to get it working with a `memcpy` for now.

Reply:
> I'm assuming you mean `gcry_sexp_length`?

Sorry, I should write it clearly. I meant `length = gcry_sexp_sprint(privkey, {0,1,2,3}, null, 0)`. `gcry_sexp_length` doesn't seem to work here.